rfd/0098-securing-docker-images.md

```diff
@@ -0,0 +1,181 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: draft
+---
+
+# RFD 0098 - Delivering secure Teleport OCI Images
+
+# Required Approvers
+* Engineering: @r0mant
+* Security: @reed
+* Product: (@xinding33 || @klizhentas)
+
+## What
+
+This RFD discusses structures and processes to increase the security of the OCI 
+images we use to deliver Teleport to our customers.
+
+## Why
+
+One of our shipping channels is a Docker image. As the provider of that docker
+image we are (at least partialy) responsible for everything in it, not just
+Teleport. We should not ship vulnerabilities to our clients, even if those 
+vulnerabilities are not directly in our software.
+
+## Details
+
+### Goals
+
+For the sake of this RFD, I will define _delivering a secure image_ as 
+
+> Reliably producing an OCI image that:
+>
+>
+>  1. a priori is unlikely to contain vulnerabilities in and of itself, 
+>  2. contains software of known provenance,
+>  3. has no warnings or vulnerabilities flagged when run through a reputable 
+>     scanner at the time of creation, and
+>  4. flags vulnerabilities even after creation, either in Teleport itself or 
+>     a dependency.
+
+### Non-goals
+
+This RFD will not discuss building images for and/or running Teleport on
+platforms other than Linux.
+
+### Approach
+
+This RFD describes a 2-pronged approach for meeting the above goals:
+
+   1. Switching to distroless base images, and 
+   2. Automated, ongoing monitoring of images using a scanning service
```

Comment:
3. Producing new images that resolve any vulnerabilities in a timely fashion to meet our vulnerability management policy (maximum length of time for fixes -- critical - 7 days, high - 30 days, moderate - 90 days, low - no set timeline (but preferably 180 days)).